src/workerd/io/io-context.c++

```diff
@@ -1304,11 +1310,33 @@ void IoContext::throwNotCurrentJsError(kj::Maybe<const std::type_info&> maybeTyp
 
 jsg::JsObject IoContext::getPromiseContextTag(jsg::Lock& js) {
   if (promiseContextTag == kj::none) {
-    promiseContextTag = jsg::JsRef(js, js.obj());
+    auto deferral = kj::heap<IoCrossContextExecutor>(kj::atomicAddRef(*deleteQueue));
+    promiseContextTag = jsg::JsRef(js, js.opaque(kj::mv(deferral)));
   }
   return KJ_REQUIRE_NONNULL(promiseContextTag).getHandle(js);
 }
 
+kj::Promise<void> IoContext::startDeleteQueueSignalTask(IoContext* context) {
+  // The promise that is returned is held by the IoContext itself, so when the
+  // IoContext is destroyed, the promise will be canceled and the loop will
+  // end. On each iteration of the loop we want to reset the cross thread
+  // signal in the delete queue, then wait on the promise. Once the promise
+  // is fulfilled, we will run an empty task to prompt the IoContext to drain
+  // the DeleteQueue.
+  try {
+    for (;;) {
+      co_await context->deleteQueue->resetCrossThreadSignal();
+      co_await context->run([](auto&) {});
+    }
+  } catch (...) {
+    auto exception = kj::getCaughtExceptionAsKj();
+    KJ_IF_SOME(a, context->actor) {
+      a.shutdownActorCache(exception);
+    }
+    kj::throwFatalException(kj::mv(exception));
```

Comment:
I guess this exception is never handled anywhere since the caller just stores the promise without ever waiting on it.

We probably need to log the exception instead of throwing it here? Or maybe `abort()`ing with it will be sufficient to make it get logged somewhere...